walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,86 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    List<Event> overlapEvents = new ArrayList<Event>();
+    List<TimeRange> overlapRanges = new ArrayList<TimeRange>();
+    List<TimeRange> noOverlapRanges = new ArrayList<TimeRange>();
+    List<TimeRange> possibleRanges = new ArrayList<TimeRange>();
+    int requestDuration = (int) request.getDuration();
+
+    if (requestDuration > TimeRange.WHOLE_DAY.duration()) {
+      return possibleRanges;
+    }
+   
+    for (String attendee : request.getAttendees()) {
```

Comment:
The way this double for loop is constructed causes a bunch of duplicate events to be added to overlapEvents. There's a few cheap ways to get around this - for example, you could change this to do a sort of set intersection between the event's attendees and the requested attendees, or for each event you could look for an attendee in common and `break` after adding the event. (Either way, the outer for loop would iterate over events.)